From 35c712358280f784a7dc3b013bd08638d1dc8c3e Mon Sep 17 00:00:00 2001
From: Ziyuan Xu <xzy.xu@rock-chips.com>
Date: Sat, 24 Mar 2018 15:36:48 +0800
Subject: [PATCH] arm64: dts: rockchip: fixup bluetooth configuration

The clk_rtc_32k is used as crystal input for wifi & bt device on
rk3308-evb-v10 board, and it should be always enabled.

Reivse the "BT,reset-gpio" to "BT,power-gpio" which the original is not
correspond to BT_REG_ON.

Change-Id: I210e5386d93fdd3004f483ab877e590f42e7ef83
Signed-off-by: Ziyuan Xu <xzy.xu@rock-chips.com>
---
 arch/arm64/boot/dts/rockchip/rk3308-evb-v10.dtsi | 11 +++++++----
 1 file changed, 7 insertions(+), 4 deletions(-)

diff --git a/arch/arm64/boot/dts/rockchip/rk3308-evb-v10.dtsi b/arch/arm64/boot/dts/rockchip/rk3308-evb-v10.dtsi
index 134d800ea6b9..9b6e672e6301 100644
--- a/arch/arm64/boot/dts/rockchip/rk3308-evb-v10.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3308-evb-v10.dtsi
@@ -163,14 +163,11 @@
 
 	wireless-bluetooth {
 		compatible = "bluetooth-platdata";
-		clocks = <&cru SCLK_RTC32K>;
-		clock-names = "ext_clock";
 		uart_rts_gpios = <&gpio4 RK_PA7 GPIO_ACTIVE_LOW>;
 		pinctrl-names = "default", "rts_gpio";
 		pinctrl-0 = <&uart4_rts>;
 		pinctrl-1 = <&uart4_rts_gpio>;
-		BT,reset_gpio    = <&gpio4 RK_PB3 GPIO_ACTIVE_HIGH>;
-		BT,wake_gpio     = <&gpio4 RK_PB2 GPIO_ACTIVE_HIGH>;
+		BT,power_gpio    = <&gpio4 RK_PB3 GPIO_ACTIVE_HIGH>;
 		BT,wake_host_irq = <&gpio4 RK_PB4 GPIO_ACTIVE_HIGH>;
 		status = "okay";
 	};
@@ -389,6 +386,12 @@
 	};
 };
 
+&uart4 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&uart4_xfer &uart4_cts>;
+	status = "okay";
+};
+
 &usb20_otg {
 	status = "okay";
 };
-- 
2.35.3

